File: backend/app/schemas/billing.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class BillingSummary(BaseModel):
    total_credits: Decimal
    used_credits: Decimal
    remaining: Decimal


class BillingRecordItem(BaseModel):
    id: int
    task_id: str | None = None
    task_name: str | None = None
    resource_type: str
    gpu_model: str | None = None
    duration_seconds: int
    duration_display: str
    cost: Decimal
    created_at: datetime


class BillingRecordsPage(BaseModel):
    items: list[BillingRecordItem]
    total: int
    page: int
    page_size: int

File: frontend/src/api/billing.ts
import request from './request'

export interface BillingSummary {
  total_credits: string;
  used_credits: string;
  remaining: string;
}

export interface BillingRecordItem {
  id: number;
  task_id?: string | null;
  task_name?: string | null;
  resource_type: string;
  gpu_model?: string | null;
  duration_seconds: number;
  duration_display: string;
  cost: string;
  created_at: string
}

export interface BillingRecordsPage {
  items: BillingRecordItem[];
  total: number;
  page: number;
  page_size: number;
}

export async function fetchBillingSummary() {
  return request.get<unknown, BillingSummary>('/billing/summary')
}

export async function fetchBillingRecords(page = 1, pageSize = 20) {
  return request.get<unknown, BillingRecordsPage>('/billing/records', {
    params: { page, page_size: pageSize },
  })
}
